# Analysis

## Revenue by year

In [1]:
## Connect

In [2]:
import os
import pandas as pd
from dotenv import load_dotenv
from sqlalchemy import create_engine

load_dotenv("../.env")
engine = create_engine(os.getenv("DATABASE_URL"))

In [3]:
pd.read_sql("SELECT order_year, SUM(order_revenue) AS revenue FROM orders WHERE order_status = 'Completed' GROUP BY order_year ORDER BY order_year", engine)

,order_year,revenue
0,2024,72394775.0
1,2025,82889275.0
2,2026,57805700.0


In [4]:
## Revenue by product category

In [5]:
pd.read_sql("SELECT product_category, SUM(order_revenue) AS revenue FROM orders WHERE order_status = 'Completed' GROUP BY product_category ORDER BY revenue DESC", engine)

,product_category,revenue
0,Electronics,37334775.0
1,Sports,36636900.0
2,Beauty,36395950.0
3,Fashion,34976025.0
4,Home & Kitchen,34485300.0
5,Groceries,33260800.0


## Revenue by customer segment

In [6]:
pd.read_sql("SELECT customer_segment, SUM(order_revenue) AS revenue FROM orders WHERE order_status = 'Completed' GROUP BY customer_segment ORDER BY revenue DESC", engine)

,customer_segment,revenue
0,Regular,72512975.0
1,Premium,64356775.0
2,Occasional,41040500.0
3,Unknown,18399150.0
4,Business,16780350.0


## Revenue by sales channel

In [7]:
pd.read_sql("SELECT sales_channel, SUM(order_revenue) AS revenue FROM orders WHERE order_status = 'Completed' GROUP BY sales_channel ORDER BY revenue DESC", engine)

,sales_channel,revenue
0,Website,49708350.0
1,Mobile App,45237525.0
2,Store,44171050.0
3,Marketplace,35224050.0
4,Instagram,23318075.0
5,Unknown,15430700.0


## Delivery performance

In [8]:
pd.read_sql("SELECT delivery_status, COUNT(*) AS orders FROM orders GROUP BY delivery_status ORDER BY orders DESC", engine)

,delivery_status,orders
0,Delivered,2210
1,In Transit,892
2,Unknown,794
3,Delayed,589
4,Returned,515


## Key findings
- Revenue rose from ₦72.4M (2024) to ₦82.9M (2025). 2026 only covers January to August.
- Electronics is the top category, but all six categories are close.
- Regular customers bring in the most revenue.
- Website is the strongest sales channel; Instagram is the weakest.
- 589 orders were delayed out of 5,000.